File: Android-Java/app/src/main/java/io/supernova/spacebook/fragment/MessagesActivity.java
/**
 *  Created by Supernova.
 */

package io.supernova.spacebook.fragment;

import android.support.v4.app.Fragment;
import android.view.View;
import io.supernova.spacebook.adapter.MessagesActivityMessagesRecyclerViewAdapter;
import io.supernova.spacebook.activity.*;
import io.supernova.spacebook.R;
import io.supernova.spacebook.adapter.MessagesActivityOnlineRecyclerViewAdapter;
import io.supernova.spacebook.databinding.MessagesActivityBinding;
import android.support.v7.widget.LinearLayoutManager;
import android.databinding.DataBindingUtil;
import io.supernova.uitoolkit.recycler.GridSpacingItemDecoration;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.app.AppCompatActivity;
import android.view.LayoutInflater;
import java.util.*;
import android.os.Bundle;
import android.view.ViewGroup;


public class MessagesActivity extends Fragment {
	
	public static MessagesActivity newInstance() {
	
		MessagesActivity fragment = new MessagesActivity();
		Bundle arguments = new Bundle();
		fragment.setArguments(arguments);
		return fragment;
	}
	
	private MessagesActivityBinding binding;
	
	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
	
		binding = DataBindingUtil.inflate(inflater, R.layout.messages_activity, container, false);
		return binding.getRoot();
	}
	
	@Override
	public void onViewCreated(View view, Bundle savedInstanceState) {
	
		super.onViewCreated(view, savedInstanceState);
		init();
	}
	
	public void init() {
	
		// Configure Online component
		binding.onlineRecyclerView.setLayoutManager(new GridLayoutManager(this.getContext(), 1, LinearLayoutManager.HORIZONTAL, false));
		binding.onlineRecyclerView.setAdapter(new MessagesActivityOnlineRecyclerViewAdapter());
		binding.onlineRecyclerView.addItemDecoration(new GridSpacingItemDecoration(this.getContext().getResources().getDimension(R.dimen.messages_activity_online_recycler_view_spacing)));
		
		// Configure Messages component
		binding.messagesRecyclerView.setLayoutManager(new LinearLayoutManager(this.getContext(), LinearLayoutManager.VERTICAL, false));
		binding.messagesRecyclerView.setAdapter(new MessagesActivityMessagesRecyclerViewAdapter());
	}
}

File: Android-Kotlin/app/src/main/java/io/supernova/spacebook/fragment/MessagesActivity.kt
/**
 * Created by Supernova.
 */

package io.supernova.spacebook.fragment

import io.supernova.spacebook.activity.*
import io.supernova.spacebook.databinding.MessagesActivityBinding
import io.supernova.spacebook.adapter.MessagesActivityMessagesRecyclerViewAdapter
import android.os.Bundle
import android.view.View
import android.databinding.DataBindingUtil
import android.support.v7.widget.LinearLayoutManager
import android.view.LayoutInflater
import android.support.v7.widget.GridLayoutManager
import java.util.*
import io.supernova.uitoolkit.recycler.GridSpacingItemDecoration
import android.support.v7.app.AppCompatActivity
import io.supernova.spacebook.R
import android.support.v4.app.Fragment
import io.supernova.spacebook.adapter.MessagesActivityOnlineRecyclerViewAdapter
import android.view.ViewGroup


class MessagesActivity: Fragment() {

	companion object {
		
		fun newInstance(): MessagesActivity {
		
			val fragment = MessagesActivity()
			val arguments = Bundle()
			fragment.arguments = arguments
			return fragment
		}
	}
	
	private lateinit var binding: MessagesActivityBinding
	
	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
	
		binding = DataBindingUtil.inflate(inflater, R.layout.messages_activity, container, false)
		return binding.root
	}
	
	override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
	
		super.onViewCreated(view, savedInstanceState)
		init()
	}
	
	fun init() {
	
		// Configure Online component
		binding.onlineRecyclerView.layoutManager = GridLayoutManager(this.context!!, 1, LinearLayoutManager.HORIZONTAL, false)
		binding.onlineRecyclerView.adapter = MessagesActivityOnlineRecyclerViewAdapter()
		binding.onlineRecyclerView.addItemDecoration(GridSpacingItemDecoration(this.context!!.resources.getDimension(R.dimen.messages_activity_online_recycler_view_spacing)))
		
		// Configure Messages component
		binding.messagesRecyclerView.layoutManager = LinearLayoutManager(this.context!!, LinearLayoutManager.VERTICAL, false)
		binding.messagesRecyclerView.adapter = MessagesActivityMessagesRecyclerViewAdapter()
	}
}
